encodings/sparse/src/compute/search_sorted.rs

```diff
@@ -23,7 +23,8 @@ impl SearchSortedFn<&SparseArray> for SparseEncoding {
             {
                 Ordering::Less => array.patches().search_sorted(value.clone(), side),
                 Ordering::Equal => Ok(SearchResult::Found(if side == SearchSortedSide::Left {
-                    array.patches().max_index()?
+                    // In case of patches being at the beginning we want the first index after the end of patches
+                    array.patches().indices().len()
```

Comment:
There was a bug here @a10y, courtesy of the fuzzer and the thing that masked it was a buggy test case